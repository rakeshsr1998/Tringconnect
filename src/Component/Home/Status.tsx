import {FlatList, Image, Pressable, Text, View} from 'react-native';
import React, {useState} from 'react';
import {styles} from './status-style';
import Modal from 'react-native-modal';

function useForceUpdate() {
  const [value, setValue] = useState(false); // integer state
  return () => setValue(!value); // update the state to force render
}

export const Status = ({status}: any) => {
  const forceUpdate = useForceUpdate();
  const [isModalVisible, setModalVisible] = useState(false);
  const [imageSrc, setImageSrc] = useState('');

  const onClick = (index: any) => {
    setImageSrc(status[index].image);
    setModalVisible(true);
    status.map((_item: any, ind: any) => {
      if (ind === index) {
        _item.isView = true;
      }
    });
    forceUpdate();
  };

  const render = ({item, index}: any) => {
    const testId = `modal-${index}`;
    if (item.addImage) {
      return (
        <Pressable
          style={{
            justifyContent: 'center',
          }}>
          <Image
            style={styles.plusIcon}
            source={require('../../assets/plus.png')}
          />
        </Pressable>
      );
    }
    return (
      <Pressable onPress={() => onClick(index)} testID={testId}>
        <View style={styles.card}>
          <View
            style={[
              styles.cardContainer,
              item.isView ? {borderColor: '#5BBC2E'} : {borderColor: '#979797'},
            ]}>
            <Image source={{uri: item.image}} style={styles.image} />
          </View>
          <Text style={styles.name}>{item.name}</Text>
          {!item.isView && <View style={styles.dot} />}
        </View>
      </Pressable>
    );
  };
  return (
    <>
      <View style={styles.container}>
        <FlatList
          ItemSeparatorComponent={() => <View style={{width: 1}} />}
          keyExtractor={({}, index) => index.toString()}
          horizontal={true}
          data={status}
          renderItem={render}
        />
      </View>
      <Modal
        onBackdropPress={() => setModalVisible(false)}
        hasBackdrop={true}
        isVisible={isModalVisible}>
        <View>
          <Pressable
            onPress={() => setModalVisible(false)}
            style={styles.closeIcon}>
            <Image
              source={require('../../assets/close.png')}
              style={styles.more}
            />
          </Pressable>
          <Image
            resizeMode="contain"
            source={{uri: imageSrc}}
            style={styles.modalCover}
          />
        </View>
      </Modal>
    </>
  );
};
