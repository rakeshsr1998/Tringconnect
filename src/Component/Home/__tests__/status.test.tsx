import {fireEvent, render, RenderAPI} from '@testing-library/react-native';
import React from 'react';
import {Status} from '../Status';

const mock1 = [
  {addImage: true},
  {
    image:
      'https://shotkit.com/wp-content/uploads/2021/06/cool-profile-pic-matheus-ferrero.jpeg',
    name: 'HR Team',
    isView: false,
  },
];

describe('<Header>', () => {
  const mockFunction = jest.fn();

  it('Should render header with home click', async () => {
    const nextButton = render(<Status status={mock1} />).getByTestId('modal-1');
    fireEvent(nextButton, 'onPress');
    expect(mockFunction).toMatchSnapshot();
  });
});
