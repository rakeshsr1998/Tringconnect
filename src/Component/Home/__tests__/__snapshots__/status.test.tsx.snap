// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> Should render header with home click 1`] = `[MockFunction]`;

exports[`<Header> Should render header with no data 1`] = `
Array [
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "height": 123,
        "justifyContent": "center",
        "padding": 10,
        "paddingRight": 0,
      }
    }
  >
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "addImage": true,
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/plus.png",
                }
              }
              style={
                Object {
                  "height": 70,
                  "width": 70,
                }
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>,
  <Modal
    hasBackdrop={true}
    isVisible={false}
    onBackdropPress={[Function]}
  >
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 15,
            "height": 30,
            "justifyContent": "center",
            "width": 30,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/close.png",
            }
          }
          style={
            Object {
              "height": 24,
              "width": 24,
            }
          }
        />
      </View>
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "",
          }
        }
        style={
          Object {
            "height": 200,
            "width": "100%",
          }
        }
      />
    </View>
  </Modal>,
]
`;
